kernel/arch/riscv/debug.c

```diff
@@ -13,27 +13,192 @@
 
 #include <elf.h>
 
+static bool fp_is_valid(ulong fp, ulong sp)
+{
+   ulong low = sp + 2 * sizeof(void *);
+   ulong high = (sp + (KERNEL_STACK_SIZE - 1)) & ~(KERNEL_STACK_SIZE - 1);
+
+   if (!fp || fp < low || fp > high || fp & 0x7)
+      return false;
+   else
+      return true;
+}
+
+static size_t
+stackwalk_riscv(void **frames,
```

Comment:
@aenrbes Thanks for the fixes!

Note: here the parameters should be aligned like this:
```
stackwalk_riscv(void **frames,
                size_t count,
                void *fp,
                pdir_t *pdir)
```

But I'm going to merge this PR anyway to unblock you and move forward. You can make this cosmetic fix in your next PR.